Implement worker side input schema generation (#166)
Azure/azure-functions-mcp-extension: src/release_notes.md

- Added support for collection/array property types in fluent tool definition APIs (#128)
- Added support for enum property types (#131)
- Updating McpInputConversionHelper to handle string target conversions (#145)

> - Adding input schema generation in the worker and enabling `UseWorkerInputSchema` to be true (#166)

Reviewer: I think the release notes should have reset here - might want to rebase!

```suggestion
- Generate the tool input schema in the worker (#166)
```
